src/devices/Card/Icode/IcodeCard.cs

```diff
@@ -0,0 +1,355 @@
+        public byte Target { get; set; }
+
+        /// <summary>
+        /// The command to execute on the card
+        /// </summary>
+        public IcodeCardCommand Command { get; set; }
+
+        /// <summary>
+        /// unique identifier of the card
+        /// </summary>
+        public byte[]? Uid { get; set; }
+
+        /// <summary>
+        /// The storage capacity
+        /// </summary>
+        public IcodeCardCapacity Capacity { get; set; }
+
+        /// <summary>
+        /// The block number to read or write
+        /// </summary>
+        public byte BlockNumber { get; set; }
+
+        /// <summary>
+        /// The block count when read multiple blocks
+        /// </summary>
+        public byte BlockCount { get; set; }
+
+        /// <summary>
+        /// The Data which has been read or to write for the specific block
+        /// </summary>
+        public byte[] Data { get; set; } = new byte[0];
+
+        /// <summary>
+        /// Constructor for IcodeCard
+        /// </summary>
+        /// <param name="rfid">A card transceiver class</param>
+        /// <param name="target">The target number as some card readers attribute one</param>
+        public IcodeCard(CardTransceiver rfid, byte target)
+        {
+            _rfid = rfid;
+            Target = target;
+            _logger = this.GetCurrentClassLogger();
+        }
+
+        /// <summary>
+        /// Provide a calculation of CRC for ISO15693
+        /// </summary>
+        /// <param name="buffer">The buffer to process</param>
+        /// <param name="crc">The CRC, Must be a 2 bytes buffer</param>
+        public void CalculateCrc15693(ReadOnlySpan<byte> buffer, Span<byte> crc)
+        {
+            if (crc.Length != 2)
+            {
+                throw new ArgumentException($"Value must be 2 bytes.", nameof(crc));
+            }
+
+            ushort polynomial = 0x8408;
+            ushort currentCrc = 0xFFFF;
+            // ISO15693-3.pdf
+            for (int i = 0; i < buffer.Length; i++)
+            {
+                currentCrc = (ushort)(currentCrc ^ buffer[i]);
+                for (int j = 0; j < 8; j++)
+                {
+                    if ((currentCrc & 0x0001) != 0)
+                    {
+                        currentCrc = (ushort)((currentCrc >> 1) ^ polynomial);
+                    }
+                    else
+                    {
+                        currentCrc = (ushort)(currentCrc >> 1);
+                    }
+                }
+            }
+
+            currentCrc = (ushort)~currentCrc;
+            crc[0] = (byte)(currentCrc & 0xFF);
+            crc[1] = (byte)((currentCrc >> 8) & 0xFF);
+        }
+
+        /// <summary>
+        /// Run the last setup command. In case of reading bytes, they are automatically pushed into the Data property
+        /// </summary>
+        /// <returns>-1 if the process fails otherwise the number of bytes read</returns>
+        public int RunIcodeCardCommand()
```

Comment:
Is this intended to be public? This requires `Command` and other things to be set to run correctly